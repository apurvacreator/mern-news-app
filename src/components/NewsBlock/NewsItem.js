import React from 'react';
import PropTypes from 'prop-types';
import { Card, SafeAnchor } from 'react-bootstrap';
import NoImage from '../../no-camera.svg';

const NewsItem = ({ urlToImage, title, url, date }) => (
    <Card style={{ width: '18rem' }}>
        <Card.Img variant="top" src={urlToImage || NoImage} alt={title} />
        <Card.Body>
            <Card.Title>{title}</Card.Title>
            <Card.Text>{new Date(date).toLocaleDateString()}</Card.Text>
            <SafeAnchor variant="primary" href={url} target='_blank'>Read the article</SafeAnchor>
        </Card.Body>
    </Card>
)

NewsItem.propTypes = {
    urlToImage: PropTypes.string,
    title: PropTypes.string,
    url: PropTypes.string,
    date: PropTypes.string,
}

export default NewsItem;