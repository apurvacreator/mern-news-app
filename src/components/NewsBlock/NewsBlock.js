import React from 'react';
import PropTypes from 'prop-types';
import NewsItem from './NewsItem';
import { Row, Col } from 'react-bootstrap';
import Info from '../Info/Info';

const NewsBlock = ({ newsArticles, isHeadlines }) => (
    <>
        <Row>
            <Col>
                {newsArticles.length ?
                    <>
                        <h2>{isHeadlines
                            ? 'Today\'s Headlines'
                            : 'Your search results'
                        }</h2>
                        <hr />
                    </>
                    : <Info
                        variant={'info'}
                        message={'No results found'} />}
            </Col>
        </Row>
        <Row>
            {newsArticles.map(({ title, url, urlToImage, publishedAt }) =>
                <Col key={title} style={{ margin: 10 }}>
                    <NewsItem
                        title={title}
                        url={url}
                        urlToImage={urlToImage}
                        date={publishedAt}
                    />
                </Col>
            )}
        </Row>
    </>
)

NewsBlock.propTypes = {
    newsArticles: PropTypes.arrayOf(PropTypes.shape({
        author: PropTypes.string,
        content: PropTypes.string,
        description: PropTypes.string,
        publishedAt: PropTypes.string,
        source: PropTypes.shape({
            id: PropTypes.string,
            name: PropTypes.string
        }),
        title: PropTypes.string,
        url: PropTypes.string,
        urlToImage: PropTypes.string
    })),
    isHeadlines: PropTypes.bool.isRequired
}

export default NewsBlock;