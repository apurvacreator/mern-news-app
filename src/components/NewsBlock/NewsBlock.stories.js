import React from 'react';
import NewsBlock from './NewsBlock';

const props = {
    newsArticles: [{
        urlToImage: "https://cbsnews3.cbsistatic.com/hub/i/r/2020/08/12/620d4e6e-bcfa-45af-8599-ba9972b6f17d/thumbnail/1200x630/7c2070a64901e2f363b1d75058f86ce8/gettyimages-1180699958.jpg",
        title: 'This is a test news',
        url: 'https://www.cbsnews.com/news/big-12-college-football-2020-season-big-ten-pac-12-postponements/',
        date: '2020-08-13T11:46:00Z'
    }, {
        urlToImage: "https://cbsnews3.cbsistatic.com/hub/i/r/2020/08/12/620d4e6e-bcfa-45af-8599-ba9972b6f17d/thumbnail/1200x630/7c2070a64901e2f363b1d75058f86ce8/gettyimages-1180699958.jpg",
        title: 'This is a test news',
        url: 'https://www.cbsnews.com/news/big-12-college-football-2020-season-big-ten-pac-12-postponements/',
        date: '2020-08-13T11:46:00Z'
    }, {
        urlToImage: "https://cbsnews3.cbsistatic.com/hub/i/r/2020/08/12/620d4e6e-bcfa-45af-8599-ba9972b6f17d/thumbnail/1200x630/7c2070a64901e2f363b1d75058f86ce8/gettyimages-1180699958.jpg",
        title: 'This is a test news',
        url: 'https://www.cbsnews.com/news/big-12-college-football-2020-season-big-ten-pac-12-postponements/',
        date: '2020-08-13T11:46:00Z'
    }],
    isHeadlines: false
}
export default {
    title: 'NewsBlock',
    component: NewsBlock,
};

export const Block = () => (
    <NewsBlock
        {...props}
    />

)
