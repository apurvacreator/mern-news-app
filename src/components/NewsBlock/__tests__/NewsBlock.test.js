import React from 'react'
import { mount } from 'enzyme'
import NewsBlock from '../NewsBlock'
import NewsItem from '../NewsItem';

const props = {
    newsArticles: [{
        urlToImage: "https://cbsnews3.cbsistatic.com/hub/i/r/2020/08/12/620d4e6e-bcfa-45af-8599-ba9972b6f17d/thumbnail/1200x630/7c2070a64901e2f363b1d75058f86ce8/gettyimages-1180699958.jpg",
        title: 'This is a test news',
        url: 'https://www.cbsnews.com/news/big-12-college-football-2020-season-big-ten-pac-12-postponements/',
        date: '2020-08-13T11:46:00Z'
    }],
    isHeadlines: false
}

describe('NewsBlock', () => {
    let wrapper
    beforeEach(() => {
        wrapper = mount(<NewsBlock {...props} />)
    })
    it('should render', () => {
        expect(wrapper).toBeTruthy()
    })
    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    })
    it('should render NewsItem component', () => {
        expect(wrapper.prop('newsArticles')).toEqual(props.newsArticles)
        expect(wrapper.find(NewsItem)).toHaveLength(1)
    })
})