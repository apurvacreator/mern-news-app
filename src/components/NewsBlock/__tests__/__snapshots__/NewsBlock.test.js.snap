// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsBlock should match snapshot 1`] = `"<div class=\\"row\\"><div class=\\"col\\"><h2>Your search results</h2><hr></div></div><div class=\\"row\\"><div style=\\"margin: 10px auto;\\" class=\\"col\\"><div style=\\"width: 18rem;\\" class=\\"card\\"><img class=\\"card-img-top\\" src=\\"https://cbsnews3.cbsistatic.com/hub/i/r/2020/08/12/620d4e6e-bcfa-45af-8599-ba9972b6f17d/thumbnail/1200x630/7c2070a64901e2f363b1d75058f86ce8/gettyimages-1180699958.jpg\\" alt=\\"This is a test news\\"><div class=\\"card-body\\"><div class=\\"card-title h5\\">This is a test news</div><p class=\\"card-text\\">Invalid Date</p><a variant=\\"primary\\" href=\\"https://www.cbsnews.com/news/big-12-college-football-2020-season-big-ten-pac-12-postponements/\\" target=\\"_blank\\">Read the article</a></div></div></div></div>"`;
